src/Microsoft.DncEng.SecretManager/Commands/ValidateAllCommand.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DncEng.CommandLineLib;
+using Microsoft.DncEng.SecretManager.StorageTypes;
+using Mono.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Command = Microsoft.DncEng.CommandLineLib.Command;
+
+namespace Microsoft.DncEng.SecretManager.Commands
+{
+    [Command("validate-all")]
+    public class ValidateAllCommand : Command
+    {
+        private readonly IConsole _console;
+        private readonly SettingsFileValidator _settingsFileValidator;
+        private readonly StorageLocationTypeRegistry _storageLocationTypeRegistry;
+        private readonly List<string> _manifestFiles = new List<string>();
+        private string _basePath;
+
+        public ValidateAllCommand(IConsole console, SettingsFileValidator settingsFileValidator, StorageLocationTypeRegistry storageLocationTypeRegistry)
+        {
+            _console = console;
+            _settingsFileValidator = settingsFileValidator;
+            _storageLocationTypeRegistry = storageLocationTypeRegistry;
+        }
+
+        public override OptionSet GetOptions()
+        {
+            return new OptionSet
+            {
+                {"m|manifest-file=", "A secret manifest file. Can be specified more than once.", m => _manifestFiles.Add(m)},
+                {"base-path=", "The base path to search for settings files.", b => _basePath = b},
+            };
+        }
+
+        public override bool AreRequiredOptionsSet()
+        {
+            return !string.IsNullOrEmpty(_basePath) || _manifestFiles.Any();
```

Comment:
Shouldn't be both required? I think Directory.EnumerateFiles fails if path is empty.

Reply:
Yes, thats correct. I messed up the boolean logic